pkg/controllers/gateway/gateway_controller.go

```diff
@@ -412,7 +389,7 @@ func (r *GatewayReconciler) reconcileTLS(ctx context.Context, upstreamGateway *g
 			downstreamSecret.Namespace = gateway.Namespace
 			downstreamSecret.Labels = secret.Labels
 			downstreamSecret.Annotations = secret.Annotations
-			accessor.AddTLS(mh.Host, downstreamSecret)
+			accessor.AddTLS(host, downstreamSecret)
 			tlsSecrets = append(tlsSecrets, downstreamSecret)
```

Comment:
There's a  few additions here related to listeners. Are we making these changes in this PR?

Reply:
yeah I think without these we would end up leaving the host service around. As part of the TLSPolicy work, we will likely move quite a bit of this logic also